repository/user_repository.py

```diff
@@ -21,4 +21,20 @@ def delete_user(self, email: str) -> None:
         user = self.get_user_by_email(email)
         if user:
             self.db.delete(user)
-            self.db.commit()
\ No newline at end of file
+            self.db.commit()
+    
+    def assign_mentor(self, mentor_email: str, mentee_email: str) -> None:
+        mentee = self.get_user_by_email(mentee_email)
+        mentor = self.get_user_by_email(mentor_email)
+        if mentee and mentor:
+            mentee.mentor_email = mentor_email
+            mentee.role = "mentee"  
+            mentor.role = "mentor"  
+            self.db.commit()
+    
+    def reinitialize_mentees_mentor(self, mentor_email: str) -> None:
+        mentees = self.db.query(User).filter(User.mentor_email == mentor_email).all()
+        for mentee in mentees:
+            mentee.mentor_email = None
+            mentee.role = "null"
```

Comment:
I think the three possible values for a role, as you defined previously are: "mentee", "mentor", or None (default value), but I'm seeing that you are setting the default value to be "null" when deleting the mentor of a mentee! Is this done on purpose?

Reply:
I defined the role class in the models and used Null as variable name for None, so the role attribute will recognize Null as its default value which points to None. However there is a typo since it is Null not null.

Reply:
Make sense now. Thank you!